### ISSUE TITLE
Crate doesn't compile with ahash feature disabled

### ISSUE THREAD

**Issue by CraftSpider** [zesterer/chumsky#81]
When the ahash feature is disabled, the crate fails due to:
"error[E0412]: cannot find type `RandomState` in module `std::collections::hash_set`"

**Comment by wackbyte** [comment]
#79

**Comment by tailhook** [comment]
Also @zesterer, is there any evidence that it's needed or should be default? It's used only in error. I understand that that error structure is surely at the hot path, but commit that introduces it doesn't explain it or shows any numbers.

So since it is not used for users that use custom error type (like me)and it brings less than trivial tree of dependencies, I propose to disable it by default:
```

│   └── ahash v0.3.8
│       └── const-random v0.1.13
│           ├── const-random-macro v0.1.13 (proc-macro)
│           │   ├── getrandom v0.2.4
│           │   │   ├── cfg-if v1.0.0
│           │   │   └── libc v0.2.112
│           │   ├── lazy_static v1.4.0
│           │   ├── proc-macro-hack v0.5.19 (proc-macro)
│           │   └── tiny-keccak v2.0.2
│           │       └── crunchy v0.2.2
│           └── proc-macro-hack v0.5.19 (proc-macro)
```

**Comment by zesterer** [comment]
> Also @zesterer, is there any evidence that it's needed or should be default?

I did some benchmarking a while ago (based on the JSON parser benchmark) and there was a *very* significant difference between `ahash` and `std`'s the default hasher.

That said, I do wonder whether it's worth using something like a `VecSet` instead: the number of possible expected tokens rarely exceeds a dozen or so. A linear search through items might not be the worst thing in the world, although obviously it might rarely have pessimistic performance. I'm open to hearing about your thoughts on this!

Another option is to move `Simple` to an extension crate. For a while now I've thought that there's room for some of the more high-level, pre-packaged types to move outside of the main crate. This would also give space for adding some more 'plug-n-play' utilities, such as an error type that uses [`ariadne`](https://github.com/zesterer/ariadne).

**Comment by tailhook** [comment]
> I did some benchmarking a while ago (based on the JSON parser benchmark) and there was a very significant difference between ahash and std's the default hasher.

Okay, if it's justified then I have no problem with it.

> That said, I do wonder whether it's worth using something like a `VecSet` instead: the number of possible expected tokens rarely exceeds a dozen or so. A linear search through items might not be the worst thing in the world, although obviously it might rarely have pessimistic performance. I'm open to hearing about your thoughts on this!
> 

I have not benchmarked similar situations recently. My current use case is not performance-sensitive. So basically I have no opinion.

> Another option is to move `Simple` to an extension crate. For a while now I've thought that there's room for some of the more high-level, pre-packaged types to move outside of the main crate. This would also give space for adding some more 'plug-n-play' utilities, such as an error type that uses [`ariadne`](https://github.com/zesterer/ariadne).

I think something better than `Cheap` along the lines of the current `Simple` should be builtin. Otherwise, it will be hard to discover and find out how to make your own error (for example, ariadne-based error may include too much details on ariadne, so might be hard to figure out error merging stuff and similar).

On the other hand, having `Simple` using default hasher. And more performant and more featureful extra error crate may be very useful.

**Comment by zesterer** [comment]
Fixed by #79
